File: pytools/data/data.code.java.client/MsgTestPhp.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class MsgTestPhp
{
	private short u16;


	public MsgTestPhp()
	{

	}

	public MsgTestPhp(Packet packet)
	{
		u16 = packet.readShort();
	}

	public byte[] encode()
	{
		Packet packet = new Packet();
		packet.writeShort(u16);
		return packet.readBytes();
	}

	public byte[] getBytes()
	{
		return this.encode();
	}



	public void setU16(short u16)
	{
		this.u16 = u16;
	}
	public int getU16()
	{
		return PacketUtil.readUShort(this.u16);
	}

}

File: pytools/data/data.code.cpp.server/MsgTestPhp.cpp
#ifndef _MSG_TEST_PHP_
#define _MSG_TEST_PHP_

#include <list>

#include <pb_type.hpp>
#include <Packet.hpp>

#include <Msg.hpp>



class MsgTestPhp
{
private:
	U16 m_u16;


public:
	BYTE* Encode(U32* len)
	{
		Packet packet;
		packet.WriteU16(m_u16);
		return packet.ReadBytes(len);
	}

	MsgTestPhp()
	{

	}

	MsgTestPhp(Packet* packet)
	{
		m_u16 = packet->ReadU16();
	}

	BYTE* GetBytes(U32* len)
	{
		return Encode(len);
	}


	void SetU16(U16 u16)
	{
		m_u16 = u16;
	}
	U16 GetU16()
	{
		return m_u16;
	}

};

#endif
